components/feature/session/src/main/java/mozilla/components/feature/session/DefaultSessionStorage.kt

```diff
@@ -121,12 +121,26 @@ class DefaultSessionStorage(private val context: Context) : SessionStorage {
     }
 
     private fun serializeEngineSession(engineSession: EngineSession): JSONObject {
-        // We don't currently persist any engine session state
-        return JSONObject()
+        return JSONObject().apply {
+            engineSession.saveState().forEach({ k, v -> if (isJsonSerializable(v)) put(k, v) })
+        }
     }
 
     private fun deserializeEngineSession(engine: Engine, json: JSONObject): EngineSession {
-        // We don't currently persist any engine session state
-        return engine.createSession()
+        return engine.createSession().apply {
+            val values = mutableMapOf<String, Any>()
+            json.keys().forEach { k -> values[k] = json[k] }
+            restoreState(values)
+        }
+    }
+
+    private fun isJsonSerializable(value: Any): Boolean {
+        // For now we only persist primitive types
+        return when (value) {
+            is Number -> true
+            is Boolean -> true
+            is String -> true
+            else -> false
+        }
```

Comment:
Could be an extension function on Any (and maybe live in Ktx?) :)

Reply:
Yes, eventually :). I actually moved it from KTX as it isn't really generic. It's quite specific to the `DefaultSessionStorage` for now (and not complete). I will change the method name and file a follow-up.